src/trunnel/netinfo.trunnel

```diff
@@ -0,0 +1,21 @@
+const NETINFO_ADDR_TYPE_IPV4 = 4;
+const NETINFO_ADDR_TYPE_IPV6 = 6;
+
+struct netinfo_addr {
+  u8 addr_type IN [NETINFO_ADDR_TYPE_IPV4, NETINFO_ADDR_TYPE_IPV6];
```

Comment:
Hmmm is it possible this is missing 3 values by any chance (looking at tor-spec.txt):

```
   "Type" is one of:
      0x00 -- Hostname
      0x04 -- IPv4 address
      0x06 -- IPv6 address
      0xF0 -- Error, transient
      0xF1 -- Error, nontransient
```

Reply:
Seems also that the are mapped from `RESOLVED_TYPE_*` values. We should put a strong comment to keep them insync.

Reply:
The old code did not use any address type values other 4 and 6. I suppose they cannot realistically appear in a valid NETINFO cell during handshake?

Added a comment in 9cd116b21323c432e528de8eb02ece617290ee79.

Reply:
Code doesn't use it but doesn't mean the protocol doesn't allow it. 

This specific "netinfo addr" structure actually should be generalized (and probably put outside the netinfo context) because it is also a structure that is used in the `RELAY_RESOLVED` cell so all type need to be there.

It just happens that the NETINFO cell will only care about two of those values.